beacon-chain/rpc/validator/proposer.go

```diff
@@ -460,3 +437,35 @@ func constructMerkleProof(trie *trieutil.SparseMerkleTrie, index int, deposit *e
 	deposit.Proof = proof
 	return deposit, nil
 }
+
+func (vs *Server) packAttestations(ctx context.Context, slot uint64) ([]*ethpb.Attestation, error) {
+	ctx, span := trace.StartSpan(ctx, "validatorServer.packAttestations")
+	defer span.End()
+	st, err := vs.HeadFetcher.HeadState(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "could fetch head state")
+	}
+	if st.Slot() < slot {
```

Comment:
I wonder if this can be 
`SlotsToEpoch(st.Slot()) < SlotsToEpoch(slot)`

In other word, having the base state up to epoch boundary slot is sufficient enough